crypto/src/main/java/org/springframework/security/crypto/bcrypt/BCryptPasswordEncoder.java

```diff
@@ -51,32 +53,77 @@ public BCryptPasswordEncoder(int strength) {
 		this(strength, null);
 	}
 
+	/**
+	 * @param version the version of bcrypt, can be 2a,2b,2y
+	 */
+	public BCryptPasswordEncoder(String version) {
```

Comment:
Let's change version inputs from the user to be a inner nested enum named BCryptVersion so that an invalid version cannot be provided.